File: s3-inf-ppo/laboratoria/lab10/php/src/Notebook.php
<?php

namespace PPO\Notebook;

use PPO\Notebook\Interfaces\NotebookEntry;

final class Notebook {

	private $entries = [];

	public function save(NotebookEntry $entry): self {
		$this->entries[] = $entry;
		return $this;
	}

	public function countEntries(): int {
		return count($this->entries);
	}

	/**
	 * @return NotebookEntry[]
	 */
	public function getAll(): array {
		return $this->entries;
	}

}

File: s3-inf-ppo/laboratoria/lab10/java/src/Notebook/Notebook.java
package Notebook;

import Notebook.Interfaces.NotebookEntry;

import java.util.ArrayList;

final public class Notebook {

    private ArrayList<NotebookEntry> entries = new ArrayList<>();

    Notebook save(NotebookEntry entry) {
        entries.add(entry);
        return this;
    }

    int countEntries() {
        return entries.size();
    }

    Iterable<NotebookEntry> getAll() {
        return entries;
    }

}
